lib/views/post/widgets/post_buttons.dart

```diff
@@ -57,114 +125,170 @@ class _PostButtonsState extends State<PostButtons> {
   void initState() {
     _voteService.getVoteOfItem(widget.info.item).then((vote) {
       setState(() {
+        if (vote == Vote.up || vote == Vote.favorite) {
```

Comment:
You don't need a setState here as the initState is called before the build method

Reply:
getVoteOfItem is async and i can not await it in the init. so the init is over before the initial state is being fetched by the vote service

Reply:
Ah ups, didn't see the Future there. Makes total sense now :)